benchmarks/benchmark_slicing.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env python3
+import os
+import time
+import numpy
+from tempfile import TemporaryDirectory
+import logging
+logger = logging.getLogger(__name__)
+import h5py
+
+#Needed for mutithreading:
+from queue import Queue
+from threading import Thread, Event
+import multiprocessing
+
+
+class Reader(Thread):
+    """Thread executing tasks from a given tasks queue"""
+    def __init__(self, queue_in, queue_out, quit_event):
+        Thread.__init__(self)
+        self._queue_in = queue_in
+        self._queue_out = queue_out
+        self._quit_event = quit_event
+        self.daemon = True
+        self.start()
+
+    def run(self):
+        while not self._quit_event.is_set():
```

Comment:
Perhaps I'm missing something, but it looks like you don't need a quit event, because the poison pill breaks out of the loop by itself.

Reply:
"belt and suspenders"

Reply:
Fair enough!